src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,1191 @@
+            asserts.fail("Failed to receive a report for the event {}".format(self._expected_event))
+
+        asserts.assert_equal(res.Header.ClusterId, self._expected_cluster_id, "Expected cluster ID not found in event report")
+        asserts.assert_equal(res.Header.EventId, self._expected_event.event_id, "Expected event ID not found in event report")
+        return res.Data
+
+
+class TC_OPSTATE_BASE():
+    def setup_base(self, test_info=None, app_pipe="/tmp/chip_all_clusters_fifo_"):
+
+        asserts.assert_true(test_info is not None,
+                            "You shall define the test info!")
+
+        self.test_info = test_info
+        self.app_pipe = app_pipe
+
+        if self.test_info.cluster == Clusters.OperationalState:
+            self.device = "Generic"
+        elif self.test_info.cluster == Clusters.OvenCavityOperationalState:
+            self.device = "Oven"
+        else:
+            asserts.fail("Cluster not supported!")
+
+    def init_test(self):
+        self.is_ci = self.check_pics("PICS_SDK_CI_ONLY")
+        if self.is_ci:
+            app_pid = self.matter_test_config.app_pid
+            if app_pid == 0:
+                app_pid = get_pid("chip-all-clusters-app")
+                if app_pid is None:
+                    asserts.fail("The --app-pid flag must be set when PICS_SDK_CI_ONLY is set")
+            self.app_pipe = self.app_pipe + str(app_pid)
+
+    # Sends and out-of-band command to test-app
+    def write_to_app_pipe(self, command):
+        with open(self.app_pipe, "w") as app_pipe:
+            app_pipe.write(command + "\n")
+
+    def send_manual_or_pipe_command(self, command):
+        if self.is_ci:
+            self.write_to_app_pipe(command)
+        else:
+            input("Press Enter when done.\n")
+
+    async def send_cmd(self, endpoint, cmd, timedRequestTimeoutMs=None):
+        logging.info("\t##### Command %s" % cmd)
+
+        try:
+            return await self.send_single_cmd(cmd=cmd,
+                                              endpoint=endpoint,
+                                              timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        except InteractionModelError as e:
+            asserts.fail("Unexpected error returned: %s" % e.status)
+
+    async def send_cmd_expect_response(self, endpoint, cmd, expected_response, timedRequestTimeoutMs=None):
+        ret = await self.send_cmd(endpoint=endpoint,
+                                  cmd=cmd,
+                                  timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        asserts.assert_equal(ret.commandResponseState.errorStateID,
+                             expected_response,
+                             "Current value not according expected. Expected %d, received %d!" %
+                             (expected_response, ret.commandResponseState.errorStateID))
+
+    async def read_expect_success(self, endpoint, attribute):
+        logging.info("\t##### Read %s" % attribute)
```

Comment:
Inclusion of explicit tabs is not recommended.

Reply:
done